app/models/name/format.rb

```diff
@@ -71,13 +71,17 @@ def real_search_name
     Name.display_to_real_search(self)
   end
 
+  def stripped_text_name
+    text_name.gsub(/([."]|group|clade)/, "").strip.squeeze(" ")
+  end
+
   def self.display_to_real_text(name)
-    name.display_name.gsub(/ ^\*?\*?__ | __\*?\*?[^_\*]*$ /x, "").
-      gsub(/__\*?\*? [^_\*]* \s (#{ANY_NAME_ABBR}) \s \*?\*?__/x, ' \1 ').
+    name.display_name.gsub(/ ^\*?\*?__ | __\*?\*?[^_*]*$ /x, "").
+      gsub(/__\*?\*? [^_*]* \s (#{ANY_NAME_ABBR}) \s \*?\*?__/x, ' \1 ').
```

Comment:
Backslash for star is just not necessary inside square brackets?

Reply:
Wasn't aware of that either, but it makes sense to me when I think about it.  I've known for a long time that `-` only has a special meaning in character classes if it is actually between two characters so if I want to include '-' in an explicit character class, then I just throw it at the end of the list (e.g., [a-z_-]).  I found all of this behavior described here: https://www.regular-expressions.info/charclass.html, in the Metacharacters Inside Character Classes section.  It also describes some interesting behavior regarding ']' that I wasn't aware of.

Reply:
And I hadn't been aware of it. But RuboCop has known about it for a little while. https://docs.rubocop.org/rubocop/0.85/cops_style.html#styleredundantregexpescape